### ISSUE TITLE
NVDA exposes the underlying semantics of elements with role="switch" but not the switch role itself

### ISSUE THREAD

**Issue by Wildebrew** [nvaccess/nvda#12306]
When you use role="switch" and an `aria-checked` attribute on either a checkbox or toggle button, NVDA exposes the underlying semantics (announced "checkbox checked" or "checkbox, pressed, checked" or that of the button "button pressed". It does not announce the word "Switch" or change its announcement from "checked" to "on" or "pressed".

### Steps to reproduce:
You can visit this collection of [toggle button switch examples](https://codepen.io/ipsumlorem16/pen/OeVyyM)
or [this HTML CodePen based on a checkbox](https://codepen.io/felipefialho/pen/DIdKj) and add `role="switch"`.

### Actual behavior:
NVDA exposes the underlying semantics (toggle button or checkbox).

### Expected behavior:
NVDA announces "switch" and either "pressed/not pressed" or just "on/off".

### System configuration
Windows 10

#### NVDA installed/portable/running from source:
Latest version/desktop

#### NVDA version:
Latest

#### Windows version:
Windows 10
#### Name and version of other software in use when reproducing the issue:
Tested with latest version of Chrome/Firefox

### Comments
This is really a continuation of discussion from [NVDA issue 9187](https://github.com/nvaccess/nvda/issues/9187)
I am mostly curious if further thought has been given to this role or if it was decdied to stick with the underlying semantics of the toggle controls (checkbox/button respectively) and not bother with changing them to better fit `role="checkbox"`. Such a decision makes sense to me, I've never really seen the point in a third toggle control, but since we are building a switch component on my team we are hoping to have some background / expectation of screen reader support as grounds for decision, or even just documentation.
I'd be happy to build out full switch controls based on either implementation as a CodePen if that helps with testing.
I will also file an issue with the ARIA Authoring Practices group to see if a switch widget is on their roadmap.

**Comment by JAWS-test** [comment]
role=switch is missing at https://codepen.io/felipefialho/pen/DIdKj

**Comment by Wildebrew** [comment]
I added it to the code manually and tested, I also know how it works
from testing an intenral example.
I should create a separate CodePen example.


On 4/19/21, JAWS-test ***@***.***> wrote:
> role=switch is missing at https://codepen.io/felipefialho/pen/DIdKj
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/12306#issuecomment-822172112


-- 
Work hard. Have fun. Make history.

**Comment by Vanals** [comment]
Noticed the same issue. Would be nice have a role switch input be announced as a `switch` and changing checked/unchecked to on/off.

**Comment by feerrenrut** [comment]
Closing as a duplicate of #11310 which has more discussion.
